libs/adapters/src/trpc/utils.ts

```diff
@@ -14,12 +14,20 @@ import {
 const log = logger(import.meta);
 
 const logError = (path: string | undefined, error: TRPCError) => {
-  const msg = `${error.code}: [${error.cause?.name ?? error.name}] ${path}: ${
-    error.cause?.message ?? error.message
-  }`;
+  const errorName = error.cause?.name ?? error.name;
+  const errorMessage = error.cause?.message ?? error.message;
+  const msg = `${error.code}: [${errorName}] ${path}: ${errorMessage}`;
+  let fingerprint = `${error.code}-${errorName}-${path}`;
```

Comment:
in two lines without using `let` to be purely functional .... learning elixir 😆 
`
const issues = Array.isArray(error.cause?.issues) ? error.cause.issues.map(i => i.code) : [];
const fingerprint = [error.code,  errorName,  path, ...issues].join('-');
`

Reply:
Won't work outright because you need to check if `issues` exists in `error.cause` to avoid the type error it would normally raise.

Reply:
true... one more check